parlai/tasks/triviaqa/agents.py

```diff
@@ -24,11 +24,10 @@ class WebTeacher(DialogTeacher):
     def __init__(self, opt, shared=None):
         if not hasattr(self, 'prefix'):
             self.prefix = ''
-            if opt['datatype'].startswith('train'):
-                self.suffix = 'train'
-            else:
-                self.suffix = 'dev'
+            self.suffix = 'train' if opt['datatype'].startswith('train') else 'dev'
 
+        if "no_evidence" not in self.__dict__:
```

Comment:
hmm maybe it would be slightly better to do `if not hasattr(self, 'no_evidence')`. 

alternatively, you can set a function like `def no_evidence(self): return False` and override it in the subclasses to avoid checking if the attribute exits